### PR TITLE
fix(cli): make config relocation instructions more explicit

### LINKED ISSUE #6864: Make config file relocation instructions more explicit
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue

After upgrading from `1.1` to `1.2` I received the following message:
```
Configuration file exists at /Users/xxx/Library/Application Support/pypoetry, reusing this directory.

Consider moving configuration to /Users/xxx/Library/Preferences/pypoetry, as support for the legacy directory will be removed in an upcoming release.
```

Similar to #6854 I (naively) assumed (based on above message) that the entire directory was configuration related and therefore moved it from `~/Library/Application Support/` to `~/Library/Preferences`.

Of course this lead to poetry no longer functioning.

If an automatic move of the config file is not in the cards, at least the warning message needs to be more explicit what file(s) actually need to be moved.

### PR DESCRIPTION
**Pull request by cpate4**
Fixes #6864

# Pull Request Check List

Resolves: #6864

With regard to the boxes below, I am not sure if there are **tests** to write for this as the fix is pertains to "copy" for the deprecation message (as opposed to logic). Happy to update any **documentation** the team feels is relevant.

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by cpate4** [comment]
pre-commit.ci autofix

File: src/poetry/locations.py [hunk 1]
```diff
@@ -34,8 +34,8 @@
     if any(file.exists() for file in (auth_toml, config_toml)):
         logger.warning(
             "Configuration file exists at %s, reusing this directory.\n\nConsider"
-            " moving configuration to %s, as support for the legacy directory will be"
-            " removed in an upcoming release.",
+            " moving toml configuration files to %s, as support for the legacy"
```
Comment (Secrus) [inline src/poetry/locations.py:37]:
```suggestion
            " moving TOML configuration files to %s, as support for the legacy"
```

**Review by Secrus — CHANGES_REQUESTED** [review]

**Comment by cpate4** [comment]
Thank you, @Secrus, for the feedback!

**Comment by Secrus** [comment]
@cpate4 could you please resolve the conflict?

**Comment by Secrus** [comment]
pre-commit.ci autofix

**Review by Secrus — APPROVED** [review]

**Comment by cpate4** [comment]
@Secrus - Thank you! This is my first (ever) open source contribution. 🙂

**Pull request merged**